packages/graphiql/src/components/DocExplorer.js

```diff
@@ -142,6 +168,24 @@ export class DocExplorer extends React.Component {
 
   // Public API
   showDoc(typeOrField) {
+    if (!typeOrField) { return; }
+
+    if (this.props.onShowDoc) {
+      let typeName;
+      let fieldName;
+      if (isType(typeOrField)) {
+        typeName = typeOrField.name;
+        fieldName = '';
+      } else {
+        fieldName = typeOrField.name;
+        const navStack = this.state.navStack;
+        const lastState = navStack[navStack.length - 1];
+        typeName = lastState.name;
```

Comment:
I don't think there's any guarantee that the navStack will be in the right order for this; for example you could navigate to a field from hovering over the query document. Please extract the type name from `typeOrField` (if possible), or find a more reliable way of getting the type.

Reply:
Hmm, ok. I don't think there is a more reliable way of getting the type from `typeOrField` alone: types describe their fields, but fields contain no reference to their types.

I can update the `TypeDoc` component to include the relevant type info when calling its `onClickField` prop, if that works?